core/database_api.php

```diff
@@ -1323,3 +1287,60 @@ function db_mysql_fix_utf8( $p_string ) {
 function db_empty_result() {
 	return new ADORecordSet_empty();
 }
+
+/**
+ * Process a query string by replacing token parameters by their bound values
+ * @param string $p_query		Query string
+ * @param array $p_arr_parms	Parameter array
+ * @return string				Processed query string
+ */
+function db_format_query_log_msg( $p_query, array $p_arr_parms ) {
+	$t_lastoffset = 0;
+	$i = 0;
+	if( !empty( $p_arr_parms ) ) {
+		# For mysql, tokens are '?', and parameters are bound sequentially
+		# For pgsql, tokens are '$number', and parameters are bound by the denoted index (1-based) in the parameter array
+		# For oracle, tokens are ':string', but ADOdb treat them as sequential in appearance order, so they behave like mysql
+		$t_regex = '/(?<token>\?|\$|:)(?<index>[0-9]*)/';
+		while( preg_match( $t_regex , $p_query, $t_matches, PREG_OFFSET_CAPTURE, $t_lastoffset ) ) {
+			$t_match_param = $t_matches[0];
+			# Realign the offset returned by preg_match as it is byte-based,
+			# which causes issues with UTF-8 characters in the query string
+			# (e.g. from custom fields names)
+			$t_utf8_offset = utf8_strlen( substr( $p_query, 0, $t_match_param[1] ), mb_internal_encoding() );
+			if( $i <= count( $p_arr_parms ) ) {
+				if( db_is_pgsql() ) {
+					# For pgsql, the bound value is indexed by the parameter name
+					$t_index = (int)$t_matches['index'][0];
+					$t_value = $p_arr_parms[$t_index-1];
+				} else {
+					$t_value = $p_arr_parms[$i];
+				}
+				if( is_null( $t_value ) ) {
+					$t_replace = 'NULL';
+				} else if( is_string( $t_value ) ) {
+					$t_replace = "'" . $t_value . "'";
+				} else if( is_integer( $t_value ) || is_float( $t_value ) ) {
+					$t_replace = (float)$t_value;
+				} else if( is_bool( $t_value ) ) {
+					if( db_is_pgsql() ) {
+						$t_replace = '\'' . $t_value . '\'';
+					} else {
+						$t_replace = $t_value;
+					}
+				} else {
+					echo( 'Invalid argument type passed to query_bound(): ' . ( $i + 1 ) );
+					exit( 1 );
```

Comment:
Not part of your change, but do we really need to *exit* here ? 
Maybe a warning is sufficient, what do you think ?